### ISSUE TITLE
minetest 5.8.0 src/client/sound/sound_data.cpp missing `#include <algorithm>`

### ISSUE THREAD

**Issue by olivia-may** [minetest/minetest#14738]
### Minetest version

```true
Minetest version 5.8.0 (Linux)
Tag 5.8.0
(Minetest won't compile)
```

### Irrlicht device

X11

### Operating system and version

Artix (Most recent version / Rolling release)
gcc (GCC) 14.1.1 20240522

### Summary

The file src/client/sound/sound_data.cpp uses C++ std library functions `std::lower_bound` and `std::upper_bound`, but because `<algorithm>` is not included, the compiler throws errors saying these functions are not defined. I noticed, in 5.9.0-dev, this is fixed. Can this be fixed in 5.8.0? I was able to fix this and run minetest as usual after adding `#include <algorithm>` to the file. Minetest 5.8.0 otherwise will not compile on my system.
Here is the git diff:
```
diff --git a/src/client/sound/sound_data.cpp b/src/client/sound/sound_data.cpp
index f0cbc6dbf..d2d626e2c 100644
--- a/src/client/sound/sound_data.cpp
+++ b/src/client/sound/sound_data.cpp
@@ -25,6 +25,7 @@ with this program; ifnot, write to the Free Software Foundation, Inc.,
 #include "sound_data.h"
 
 #include "sound_constants.h"
+#include <algorithm>
 
 namespace sound {

```

### Steps to reproduce

Compile minetest 5.8.0 and wait for it to get to src/client/sound/sound_data.cpp.

**Comment by olivia-may** [comment]
I found the commit that fixes this
https://github.com/minetest/minetest/commit/c9ab61aa8cd0ca50bc8c4d90f257bb5ed3b0bb00
Maybe a new tag could be made? 5.8.2? This commit was created on the same month as 5.8.0's release, so I don't think it would be that hard.

**Comment by olivia-may** [comment]
I checked, minetest 5.8.1-android also has this bug.

**Comment by appgurueu** [comment]
Closing this since [the 5.9.0 feature freeze has begun](https://github.com/minetest/minetest/issues/13285): This will be fixed in the 5.9.0 release. There is little point in a 5.8.2 when 5.9.0 is around the corner.
